PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/EnvironmentVariablesTextFieldWithBrowseButton.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Originally copied from com/intellij/execution/configuration/EnvironmentVariablesTextFieldWithBrowseButton.java
+ */
+
+package com.microsoft.intellij.runner.springcloud.ui;
+
+import com.intellij.execution.util.EnvVariablesTable;
+import com.intellij.execution.util.EnvironmentVariable;
+import com.intellij.icons.AllIcons;
+import com.intellij.openapi.ui.TextFieldWithBrowseButton;
+import com.intellij.openapi.ui.ValidationInfo;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.ui.DocumentAdapter;
+import com.intellij.ui.UserActivityProviderComponent;
+import com.intellij.util.EnvironmentUtil;
+import com.intellij.util.containers.ContainerUtil;
+import com.microsoft.intellij.ui.components.AzureDialogWrapper;
+import net.miginfocom.swing.MigLayout;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import javax.swing.event.DocumentEvent;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class EnvironmentVariablesTextFieldWithBrowseButton extends TextFieldWithBrowseButton
```

Comment:
We can use EnvironmentVariableTable only without text field in setting edtor.

Reply:
Removed this component and use `EnvironmentVariableTable` directly